bridge-fungible-token/src/bridge_fungible_token.sw

```diff
@@ -0,0 +1,150 @@
+contract;
+
+dep data;
+dep errors;
+dep events;
+dep utils;
+
+use bridge_fungible_token_abi::BridgeFungibleToken;
+use contract_message_receiver::MessageReceiver;
+use core::num::*;
+use errors::BridgeFungibleTokenError;
+use events::{RefundRegisteredEvent, WithdrawalEvent};
+use std::{
+    chain::auth::{
+        AuthError,
+        msg_sender,
+    },
+    constants::ZERO_B256,
+    context::{
+        call_frames::{
+            contract_id,
+            msg_asset_id,
+        },
+        msg_amount,
+    },
+    logging::log,
+    storage::StorageMap,
+    vm::evm::evm_address::EvmAddress,
+};
+use utils::{
+    burn_tokens,
+    correct_input_type,
+    decompose,
+    input_message_data,
+    input_message_data_length,
+    input_message_recipient,
+    input_message_sender,
+    mint_and_transfer_tokens,
+    parse_message_data,
+    safe_b256_to_u64,
+    send_message_output,
+    transfer_tokens,
+    u64_to_b256,
+};
+
+////////////////////////////////////////
+// Storage declarations
+////////////////////////////////////////
+storage {
+    refund_amounts: StorageMap<(EvmAddress, EvmAddress), Option<b256>> = StorageMap {},
```

Comment:
Why `Option<b256>` instead of just `b256`? An option is going to consume more storage and since we're working with amounts, 0 is a perfect default anyway. In other words, we have no use case to ever set map values to `None`.